components/ExerciseCard/ExerciseCard.tsx

```diff
@@ -0,0 +1,78 @@
+import React, { useState } from 'react'
+import { NewButton } from '../theme/Button'
+import { Text } from '../theme/Text'
+
+export type ExerciseCardProps = {
+  challengeName: string
+  problem: string
+  answer: string
+  explanation: string
+  className?: string
+}
+
+const ExerciseCard = ({
+  challengeName,
+  problem,
+  answer,
+  explanation,
+  className = ''
+}: ExerciseCardProps) => {
+  const [studentAnswer, setStudentAnswer] = useState('')
+  const [message, setMessage] = useState('')
+  const [answerShown, setAnswerShown] = useState(false)
+
+  return (
+    <section className={`card p-4 border-0 shadow ${className}`}>
+      <div className="fw-bold mb-2">Problem</div>
+      <div className="d-flex mb-2">
+        <pre className="w-50 bg-light py-3 px-4 mb-0 me-3">{problem}</pre>
+        <div className="w-50 ms-3">
+          <div className="mb-2">{challengeName}</div>
+          <input
+            aria-label="User answer"
+            className="form-control mb-2"
+            value={studentAnswer}
+            onChange={e => setStudentAnswer(e.target.value)}
+          />
+          <div className={`my-2 ${message === '' ? 'invisible' : ''}`}>
+            <Text size="sm">{message === '' ? '.' : message}</Text>
```

Comment:
The reason why I put a "." here is because I wanted the text to retain its height so that there wouldn't be a content shift. There might be a better way to do this. Maybe I can add a min-height CSS property to this element.

Reply:
Yeah I agree that min-height would be a better solution than using `.`, its less hacky since we're explicitly setting the css rules

Reply:
Done.